onetrain.py

```diff
@@ -171,53 +212,56 @@ def caption_progress_callback(current, total):
     import templates
 
     if args.config:
         with open(args.config, encoding='utf-8') as f:
             # config_json = json.load(f)
             pass
-    if args.type:
+    if getattr(args, 'type', None):
         templates.config['model_type'] = "STABLE_DIFFUSION_XL_10_BASE" if args.type == 'sdxl' else 'STABLE_DIFFUSION_15'
-    if args.model:
+    if getattr(args, 'model', None):
         templates.config['base_model_name'] = args.model
-    if args.optimizer:
+    if getattr(args, 'optimizer', None):
         templates.config['optimizer']['optimizer'] = args.optimizer
-    if args.scheduler:
+    if getattr(args, 'scheduler', None):
         templates.config['learning_rate_scheduler'] = args.scheduler
-    if args.rank:
+    if getattr(args, 'rank', None):
         templates.config['lora_rank'] = args.rank
-    if args.alpha:
+    if getattr(args, 'alpha', None):
         templates.config['lora_alpha'] = args.alpha
-    if args.batch:
+    if getattr(args, 'batch', None):
         templates.config['batch_size'] = args.batch
-    if args.accumulation:
+    if getattr(args, 'accumulation', None):
         templates.config['gradient_accumulation_steps'] = args.accumulation
-    if args.resolution:
+    if getattr(args, 'resolution', None):
         templates.config['resolution'] = str(args.resolution)
-    if args.epochs:
+    if getattr(args, 'epochs', None):
         templates.config['epochs'] = args.epochs
-    if args.triton:
+    if getattr(args, 'triton', None):
         templates.config['optimizer.use_triton'] = True
-    if args.resume:
+    if getattr(args, 'resume', None):
         templates.config['continue_last_backup'] = True
-    if args.te:
+    if getattr(args, 'te', None):
         templates.config['text_encoder']['train'] = True
-    if args.bias:
+    if getattr(args, 'bias', None):
         templates.config['optimizer']['use_bias_correction'] = True
-    if args.backup:
+    if getattr(args, 'backup', None):
         templates.config['backup_after'] = int(templates.config['epochs'] / args.backup)
         templates.config['backup_after_unit'] = 'EPOCH'
-    if args.save:
+    if getattr(args, 'save', None):
         templates.config['save_after'] = int(templates.config['epochs'] / args.save)
         templates.config['save_after_unit'] = 'EPOCH'
     templates.config['debug_dir'] = os.path.join(args.tmp, 'debug')
     templates.config['workspace_dir'] = os.path.join(args.tmp, 'workspace')
     templates.config['cache_dir'] = os.path.join(args.tmp, 'cache')
     templates.config['concept_file_name'] = os.path.join(args.tmp, 'concept.json')
-    templates.config['output_model_destination'] = args.output or os.path.join(args.tmp, f'{args.concept}.safetensors')
+    if getattr(args, 'output', None):
+        templates.config['output_model_destination'] = args.output
+    else:
+        templates.config['output_model_destination'] = os.path.join(args.tmp, f'{args.concept}.safetensors')
```

Comment:
if config does not have valid output, then its not a valid config - why store train results to temp folder?

Reply:
This is how it was setup initially in deleted line 220 .  I just changed it to use getattr() function